src/Resources/ResourceManager/Implementation/ResourceGroups/ExportAzureResourceGroupCmdlet.cs

```diff
@@ -93,63 +96,90 @@ public class ExportAzureResourceGroupCmdlet : ResourceManagerCmdletBaseWithApiVe
         public SwitchParameter Force { get; set; }
 
+        /// <summary>
+        /// Gets or sets the API version.
+        /// </summary>
+        [CmdletParameterBreakingChange("ApiVersion", ChangeDescription = "Parameter is being deprecated without being replaced")]
+        [Parameter(Mandatory = false, HelpMessage = "When set, indicates the version of the resource provider API to use. If not specified, the API version is automatically determined as the latest available.")]
+        [ValidateNotNullOrEmpty]
+        public override string ApiVersion { get; set; }
+
         /// <summary>
         /// Executes the cmdlet.
         /// </summary>
         protected override void OnProcessRecord()
         {
             base.OnProcessRecord();
+            String contents;
+
             if (ShouldProcess(ResourceGroupName, VerbsData.Export))
             {
 
                 var resourceGroupId = this.GetResourceGroupId();
 
-                var apiVersion = this.ApiVersion ?? DefaultApiVersion;
-
-                var parameters = new ExportTemplateParameters
+                if (this.ApiVersion is null)
                 {
-                    Resources = this.GetResourcesFilter(resourceGroupId: resourceGroupId),
-                    Options = this.GetExportOptions(),
-                };
-
-                var operationResult = this.GetResourcesClient()
-                    .InvokeActionOnResource<JObject>(
-                        resourceId: resourceGroupId,
-                        action: Constants.ExportTemplate,
-                        parameters: parameters.ToJToken(),
-                        apiVersion: apiVersion,
-                        cancellationToken: this.CancellationToken.Value)
-                    .Result;
-
-                var managementUri = this.GetResourcesClient()
-                    .GetResourceManagementRequestUri(
-                        resourceId: resourceGroupId,
-                        apiVersion: apiVersion,
-                        action: Constants.ExportTemplate);
-
-                var activity = string.Format("POST {0}", managementUri.PathAndQuery);
-                var resultString = this.GetLongRunningOperationTracker(activityName: activity,
-                    isResourceCreateOrUpdate: false)
-                    .WaitOnOperation(operationResult: operationResult);
-
-                var template = JToken.FromObject(JObject.Parse(resultString)["template"]);
-
-                if (JObject.Parse(resultString)["error"] != null)
+                    var parameters = new Management.ResourceManager.Models.ExportTemplateRequest
+                    {
+                        Resources = this.GetResourcesFilter(resourceGroupId: resourceGroupId),
+                        Options = this.GetExportOptions(),
+                    };
+
+                    var exportedTemplate = ResourceManagerSdkClient.ExportResourceGroup(ResourceGroupName, parameters);
```

Comment:
Have you had a chance to test export async route? AFAIK, the service side goes into async operation based on the time it takes to export or the number of resources (I think it was 20). It might be easier to test async route by trying more than 20 resources to export rather than trying to hit the timebound. Do you mind testing this scenario, if you haven't already.

Reply:
I don't quite understand the scenario.

Reply:
No worries. We'll chat offline.

---
In reply to: [510976108](https://github.com/Azure/azure-powershell/pull/13275#discussion_r510976108) [](ancestors = 510976108)